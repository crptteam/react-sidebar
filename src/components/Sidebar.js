import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Blocker from '../styled/Blocker';
import Panel from '../styled/Panel';
import Wrapper from '../styled/Wrapper';

class Sidebar extends Component {
    constructor(props) {
        super(props);

        this.bodyOverflowOriginal = document.body.style.overflow;
    }

    render() {
        if (this.props.active) {
            document.body.style.overflow = 'hidden';

            return (
                <Wrapper>
                    <Blocker onClick={this.props.onClose} onTouchMove={e => e.preventDefault()}/>
                    <Panel>
                        {this.props.children}
                    </Panel>
                </Wrapper>
            );
        } else {
            document.body.style.overflow = this.bodyOverflowOriginal;
            return (null);
        }
    }
}

Sidebar.propTypes = {
    active: PropTypes.bool.isRequired,
    onClose: PropTypes.func.isRequired
};

export default Sidebar;