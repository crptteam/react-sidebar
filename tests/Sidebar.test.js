import React from 'react';
import Sidebar from '../src/';

describe('Sidebar', () => {
    it('Should renders without problems', () => {
        const sidebar = shallow(<Sidebar active={true}/>);

        expect(sidebar).toMatchSnapshot();
    });
});