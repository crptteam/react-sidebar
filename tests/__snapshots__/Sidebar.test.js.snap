// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should renders without problems 1`] = `
<Blocker
  onClick={[Function]}
>
  <Panel
    onClick={[Function]}
  >
    <CloseButton
      onClick={[Function]}
    >
      Close
    </CloseButton>
  </Panel>
</Blocker>
`;
